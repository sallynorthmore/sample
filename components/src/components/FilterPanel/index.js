import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import ChoiceList from '../ChoiceList';
import DatePicker from '../DatePicker';
import NumberPicker from '../NumberPicker';
import Filter from '../Filter';
import FilterSearch from '../FilterSearch';
import { IconChevron } from '../Icon';
import {
	Clear,
	Content,
	FilterItem,
	Filters,
	Footer,
	Header,
	Inner,
	Item,
	ItemIcon,
	Items,
	Label,
	Nav,
	Search,
	Summary,
} from './styles';

class FilterPanel extends Component {
	state = {
		searchTerm: '',
		openFilterId: null,
		isFilterOpen: false,
		isPanelClosing: false,
	};

	getQueryElementComponent({ queryElement, operation, selectedValues }) {
		const { onFilterChange } = this.props;
		switch (queryElement.elementType) {
			case 'CHOICE_LIST':
				return (
					<ChoiceList
						isMultiChoice={queryElement.maxChoices !== 1}
						choices={queryElement.choices}
						selectedValues={selectedValues}
						onValueChange={opDelta =>
							onFilterChange({
								...operation,
								...opDelta,
							})
						}
					/>
				);
			case 'DATE_RANGE_PICKER':
				return (
					<DatePicker
						startDate={operation.start}
						endDate={operation.end}
						onDateSelectionChange={opDelta =>
							onFilterChange({
								...operation,
								...opDelta,
							})
						}
					/>
				);
			case 'NUMBER':
				return (
					<NumberPicker
						title={queryElement.title}
						minimum={queryElement.minimum}
						maximum={queryElement.maximum}
						delta={queryElement.delta}
						value={operation.value ? parseFloat(operation.value) : null}
						onValueChange={opDelta =>
							onFilterChange({
								...operation,
								...opDelta,
							})
						}
					/>
				);
			default:
				return null;
		}
	}

	closeFilterPanel = () => {
		this.setState({
			isFilterOpen: false,
		});
	};

	handleDoneClick = () => {
		// Hide filters temporarily on done click
		this.setState({
			isPanelClosing: true,
		});

		this.closeFilterPanel();
		this.props.onDoneClick();

		setTimeout(() => {
			this.setState({
				isPanelClosing: false,
			});
		}, 1000);
	};

	openFilterPanel = id => {
		this.setState({
			openFilterId: id,
			isFilterOpen: true,
		});
	};

	searchFilters = searchTerm => {
		this.setState({
			searchTerm: searchTerm,
		});
	};

	render() {
		const { filters, onClearClick } = this.props;
		const {
			openFilterId,
			isPanelClosing,
			searchTerm,
			isFilterOpen,
		} = this.state;

		return (
			<Inner>
				<Content isOpen={isFilterOpen} isPanelClosing={isPanelClosing}>
					<Nav>
						<Header>
							<Clear onClick={onClearClick}>Clear all filters</Clear>

							<Search>
								<FilterSearch handleSearchChange={this.searchFilters} />
							</Search>
						</Header>
						<Items>
							{filters.map(({ queryElement, summary, selectedValues }) => (
								<Item
									key={queryElement.id}
									onClick={() => this.openFilterPanel(queryElement.id)}
									hasFilters={selectedValues.length > 0}
									isVisible={queryElement.label
										.toLowerCase()
										.includes(searchTerm)}
								>
									<Label>{queryElement.label}</Label>

									{summary && <Summary>{summary}</Summary>}

									<ItemIcon>
										<IconChevron width={10} />
									</ItemIcon>
								</Item>
							))}
						</Items>
						<Footer>
							<Button text="Done" responsive onClick={this.handleDoneClick} />
						</Footer>
					</Nav>

					<Filters>
						{filters
							.filter(({ queryElement }) => queryElement.id === openFilterId)
							.map(({ queryElement, operation, summary, selectedValues }) => {
								return (
									<FilterItem key={queryElement.id}>
										<Filter
											onFilterTitleClick={this.closeFilterPanel}
											title={queryElement.label}
											summary={summary}
										>
											{this.getQueryElementComponent({
												queryElement,
												operation,
												selectedValues,
											})}
										</Filter>
									</FilterItem>
								);
							})}
					</Filters>
				</Content>
			</Inner>
		);
	}
}

FilterPanel.propTypes = {
	filters: PropTypes.arrayOf(
		PropTypes.shape({
			queryElement: PropTypes.object.isRequired,
			operation: PropTypes.object.isRequired,
		})
	),
	onFilterChange: PropTypes.func,
	onDoneClick: PropTypes.func,
	onClearClick: PropTypes.func,
};

FilterPanel.defaultProps = {
	filters: [],
	onFilterChange: () => null,
};

export default FilterPanel;
