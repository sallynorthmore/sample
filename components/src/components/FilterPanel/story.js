// import React from 'react';
// import { storiesOf } from '@storybook/react';
// import { withInfo } from '@storybook/addon-info';
// import { action } from '@storybook/addon-actions';
// import { filterModelFactory } from '../../services/filter-models';
//
// import FilterPanel from './index';
//
// const stories = storiesOf('FilterPanel', module);
//
// const mockQueryUiElements = [
// 	{
// 		id: 'elem1',
// 		label: 'Select multiple',
// 		elementType: 'CHOICE_LIST',
// 		valueAddress: {
// 			tableViewConfigId: 'someTableViewConfigId1',
// 			operationId: 'mockOp1',
// 		},
// 		choices: [
// 			{
// 				label: 'A',
// 				value: 'a',
// 			},
// 			{
// 				label: 'B',
// 				value: 'b',
// 			},
// 			{
// 				label: 'C',
// 				value: 'c',
// 			},
// 			{
// 				label: 'X',
// 				value: 'x',
// 			},
// 			{
// 				label: 'Y',
// 				value: 'y',
// 			},
// 			{
// 				label: 'Z',
// 				value: 'z',
// 			},
// 			{
// 				label: 'K',
// 				value: 'k',
// 			},
// 			{
// 				label: 'J',
// 				value: 'j',
// 			},
// 			{
// 				label: 'I',
// 				value: 'i',
// 			},
// 			{
// 				label: 'Xx',
// 				value: 'xx',
// 			},
// 			{
// 				label: 'Yy',
// 				value: 'yy',
// 			},
// 			{
// 				label: 'Zz',
// 				value: 'zz',
// 			},
// 			{
// 				label: 'Kk',
// 				value: 'kk',
// 			},
// 			{
// 				label: 'Jk',
// 				value: 'jk',
// 			},
// 			{
// 				label: 'Ik',
// 				value: 'ik',
// 			},
// 			{
// 				label: 'XK',
// 				value: 'xk',
// 			},
// 			{
// 				label: 'YA',
// 				value: 'ya',
// 			},
// 			{
// 				label: 'ZA',
// 				value: 'za',
// 			},
// 			{
// 				label: 'KB',
// 				value: 'kb',
// 			},
// 			{
// 				label: 'JJ',
// 				value: 'jj',
// 			},
// 			{
// 				label: 'IA',
// 				value: 'ia',
// 			},
// 		],
// 	},
// 	{
// 		id: 'elem2',
// 		label: 'Sources',
// 		elementType: 'CHOICE_LIST',
// 		valueAddress: {
// 			tableViewConfigId: 'someTableViewConfigId1',
// 			operationId: 'mockOp2',
// 		},
// 		choices: [
// 			{
// 				label: 'D',
// 				value: 'd',
// 			},
// 			{
// 				label: 'E',
// 				value: 'e',
// 			},
// 		],
// 	},
// 	{
// 		id: 'someDateRangeElementId',
// 		label: 'Date filter with long text label',
// 		elementType: 'DATE_RANGE_PICKER',
// 		valueAddress: {
// 			tableViewConfigId: 'someTableViewConfigId1',
// 			operationId: 'mockOp3',
// 		},
// 		minDate: '2018-1-20',
// 		maxDate: '2018-2-18',
// 	},
// 	{
// 		id: 'someNumberPickerElementId',
// 		label: 'Min size',
// 		elementType: 'NUMBER',
// 		valueAddress: {
// 			tableViewConfigId: 'someTableViewConfigId1',
// 			operationId: 'mockOp4',
// 		},
// 		title: 'Pick a min size',
// 	},
// ];
// const mockOperations = [
// 	{
// 		operationId: 'mockOp1',
// 		operationType: 'FILTER_MULTIPLE_VALUES',
// 		values: ['b', 'c'],
// 	},
// 	{
// 		operationId: 'mockOp2',
// 		operationType: 'FILTER_MULTIPLE_VALUES',
// 		values: ['e'],
// 	},
// 	{
// 		operationId: 'mockOp3',
// 		operationType: 'FILTER_DATE_RANGE',
// 		start: '2/6/18',
// 		end: '2/12/18',
// 	},
// 	{
// 		operationId: 'mockOp4',
// 		operationType: 'NUMBER',
//
// 		minimum: { type: '0' },
// 		maximum: { type: '10000' },
// 		delta: { type: '100' },
// 	},
// ];
//
// const filterModels = mockQueryUiElements.map(queryElement =>
// 	filterModelFactory(
// 		mockOperations.find(
// 			v => v.operationId === queryElement.valueAddress.operationId
// 		),
// 		queryElement
// 	)
// );
//
// const panelProps = {
// 	filters: filterModels,
// 	onDoneClick: action('Done button click'),
// 	onClearClick: action('Clear button click'),
// };
//
// stories.add(
// 	'FilterPanel',
// 	withInfo('FilterPanel documentation')(() => {
// 		return (
// 			<div style={{ maxWidth: '336px', minHeight: '720px', display: 'flex' }}>
// 				<FilterPanel {...panelProps} />
// 			</div>
// 		);
// 	})
// );
