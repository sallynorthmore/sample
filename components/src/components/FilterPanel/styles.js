import styled from 'styled-components';

const width = 336;
const panelWidth = `${width}px`;
const easeInOut = 'cubic-bezier(0.4, 0.0, 0.2, 1)';
const accentBlue = 'rgba(0, 129, 187, 1)';

export const Inner = styled.div`
	background: white;
	border-left: 1px solid rgba(216, 216, 216, 0.8);
	border-right: 1px solid rgba(216, 216, 216, 0.8);
	display: flex;
	flex-direction: column;
	min-width: ${panelWidth};
	overflow: hidden;
	position: relative;
`;

export const Search = styled.div`
	position: relative;
`;

export const Clear = styled.button`
	background: transparent;
	border: 0 none;
	color: rgba(11, 26, 52, 1);
	cursor: pointer;
	display: inline-block;
	font: 300 13px/1 'Lato';
	justify-content: center;
	opacity: 0.3;
	outline: none;
	padding: 16px 0 0;
	position: absolute;
	right: 0;
	top: 0;
	transition: opacity 0.195s cubic-bezier(0, 0, 0.2, 1);
	user-select: none;
	width: auto;
	z-index: 1;

	&:hover {
		color: rgba(0, 129, 187, 1);
		opacity: 1;
	}

	&:disabled {
		opacity: 0;
		pointer-events: none;
	}
`;

export const Content = styled.div`
	display: flex;
	flex: 1;
	opacity: ${props => (props.isPanelClosing ? '0' : '1')};
	overflow: hidden;
	position: relative;
	transform: ${props =>
		props.isOpen ? `translateX(-${panelWidth})` : 'translateX(0)'};
	transition: transform 0.195s cubic-bezier(0.4, 0, 1, 1);
	width: ${width * 2}px;
`;

export const Nav = styled.div`
	display: flex;
	flex-direction: column;
	overflow: hidden;
	position: relative;
	width: ${panelWidth};
`;

export const Header = styled.header`
	flex: 0 0 72px;
	margin: 0 16px;
	min-height: 72px;
	padding: 24px 0 0;
	position: relative;
`;
export const Items = styled.div`
	background: white;
	flex: 1 1 0;
	margin: 0;
	overflow: auto;
	padding: 16px 0;
	position: relative;
	width: 100%;
`;

export const Item = styled.div`
	align-items: center;
	color: ${props => (props.hasFilters ? accentBlue : 'rgba(51, 51, 51, 1)')};
	cursor: pointer;
	display: ${props => (props.isVisible ? 'flex' : 'none')};

	justify-content: space-between;
	margin: 0;
	min-height: 54px;
	padding: 0 16px;
	position: relative;
	transition: background 0.195s ${easeInOut}, color 0.195s ${easeInOut};
	user-select: none;
	width: 100%;

	&:hover {
		background: rgba(248, 248, 248, 1);
		color: ${accentBlue};
	}
`;

export const ItemIcon = styled.span`
	display: inline-block;
	margin-right: 0;
	transform: rotate(-90deg);
`;

export const Label = styled.span`
	font: 400 14px/1 'Lato';
	margin-right: 4px;
	min-width: 0;
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
`;

export const Summary = styled.span`
	display: inline-block;
	font-weight: 400;
	margin: auto 16px auto auto;
	max-width: 136px;
	overflow: hidden;
	padding: 0 16px 0 0;
	position: relative;
	text-overflow: ellipsis;
	white-space: nowrap;

	&::after {
		background: rgba(226, 228, 231, 1);
		content: '';
		display: block;
		height: 10px;
		position: absolute;
		right: 0;
		top: 2px;
		width: 1px;
	}
`;

export const Footer = styled.div`
	background: white;
	flex: 0 0 72px;
	margin: auto 0 0;
	min-height: 72px;
	padding: 16px;
	width: 100%;
`;

export const Filters = styled.div`
	flex: 0 0 ${panelWidth};
	overflow: hidden;
	max-width: ${panelWidth};
	position: relative;
	width: ${panelWidth};
`;

export const FilterItem = styled.div`
	background: white;
	display: flex;
	height: 100%;
	left: 0;
	overflow: hidden;
	position: absolute;
	top: 0;
	width: 100%;
	z-index: ${props => (props.isActive ? '2' : '0')};
`;
