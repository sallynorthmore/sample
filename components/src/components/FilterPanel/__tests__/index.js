import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import FilterPanel from '../index';

const testProps = {
	filters: [],
	active: true,
	onDoneClick: () => null,
	onClearClick: () => null,
};

describe('<FilterPanel /> snapshots', () => {
	it('Expect snapshot to match without props', () => {
		const tree = renderer.create(<FilterPanel />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Expect snapshot to match with props', () => {
		const tree = renderer.create(<FilterPanel {...testProps} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
