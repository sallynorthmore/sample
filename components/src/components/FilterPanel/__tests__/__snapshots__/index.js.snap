// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterPanel /> snapshots Expect snapshot to match with props 1`] = `
.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(1,129,187,1);
  border: 0 none;
  color: white;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: 400 13px/1 Lato;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 36px;
  outline: none;
  padding: 16px;
  -webkit-transition: background 0.195s cubic-bezier(0.4,0,0.2,1);
  transition: background 0.195s cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.c11:hover,
.c11:focus {
  background: rgba(0,114,165,1);
}

.c11:active {
  background: rgba(0,114,165,1);
}

.c11:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.c6 {
  padding: 0;
  position: relative;
}

.c7 {
  border: 0;
  border-bottom: 1px solid rgba(227,232,236,1);
  font: 400 13px/ 1 'Lato';
  outline: none;
  padding: 8px 8px 16px 24px;
  width: 100%;
}

.c7::selection {
  background: rgba(0,129,187,1);
  color: white;
}

.c7::-webkit-input-placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7::-moz-placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7:-ms-input-placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7::placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7:focus {
  color: rgba(0,129,187,1);
  font-weight: 400;
}

.c8 {
  bottom: 15px;
  color: rgba(60,79,114,0.6);
  position: absolute;
  z-index: 1;
}

.c0 {
  background: white;
  border-left: 1px solid rgba(216,216,216,0.8);
  border-right: 1px solid rgba(216,216,216,0.8);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 336px;
  overflow: hidden;
  position: relative;
}

.c5 {
  position: relative;
}

.c4 {
  background: transparent;
  border: 0 none;
  color: rgba(11,26,52,1);
  cursor: pointer;
  display: inline-block;
  font: 300 13px/1 'Lato';
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.3;
  outline: none;
  padding: 16px 0 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: opacity 0.195s cubic-bezier(0,0,0.2,1);
  transition: opacity 0.195s cubic-bezier(0,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: auto;
  z-index: 1;
}

.c4:hover {
  color: rgba(0,129,187,1);
  opacity: 1;
}

.c4:disabled {
  opacity: 0;
  pointer-events: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  opacity: 1;
  overflow: hidden;
  position: relative;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 0.195s cubic-bezier(0.4,0,1,1);
  -webkit-transition: transform 0.195s cubic-bezier(0.4,0,1,1);
  transition: transform 0.195s cubic-bezier(0.4,0,1,1);
  width: 672px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  position: relative;
  width: 336px;
}

.c3 {
  -webkit-flex: 0 0 72px;
  -ms-flex: 0 0 72px;
  flex: 0 0 72px;
  margin: 0 16px;
  min-height: 72px;
  padding: 24px 0 0;
  position: relative;
}

.c9 {
  background: white;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  margin: 0;
  overflow: auto;
  padding: 16px 0;
  position: relative;
  width: 100%;
}

.c10 {
  background: white;
  -webkit-flex: 0 0 72px;
  -ms-flex: 0 0 72px;
  flex: 0 0 72px;
  margin: auto 0 0;
  min-height: 72px;
  padding: 16px;
  width: 100%;
}

.c12 {
  -webkit-flex: 0 0 336px;
  -ms-flex: 0 0 336px;
  flex: 0 0 336px;
  overflow: hidden;
  max-width: 336px;
  position: relative;
  width: 336px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <header
        className="c3"
      >
        <button
          className="c4"
          onClick={[Function]}
        >
          Clear all filters
        </button>
        <div
          className="c5"
        >
          <div
            className="c6"
          >
            <input
              className="c7"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
              value=""
            />
            <div
              className="c8"
            >
              <svg
                viewBox="0 0 17 17"
                width="15px"
              >
                <g
                  fill="none"
                  stroke="none"
                  strokeLinecap="round"
                  strokeWidth="1"
                >
                  <g
                    stroke="currentColor"
                    strokeWidth="2"
                    transform="translate(1.000000, 1.000000)"
                  >
                    <g>
                      <circle
                        cx="6.5"
                        cy="6.5"
                        r="6"
                      />
                      <path
                        d="M14.5,14.5 L10.742,10.742"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
      </header>
      <div
        className="c9"
      />
      <div
        className="c10"
      >
        <button
          className="c11"
          disabled={undefined}
          onClick={[Function]}
        >
          Done
        </button>
      </div>
    </div>
    <div
      className="c12"
    />
  </div>
</div>
`;

exports[`<FilterPanel /> snapshots Expect snapshot to match without props 1`] = `
.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(1,129,187,1);
  border: 0 none;
  color: white;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: 400 13px/1 Lato;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 36px;
  outline: none;
  padding: 16px;
  -webkit-transition: background 0.195s cubic-bezier(0.4,0,0.2,1);
  transition: background 0.195s cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.c11:hover,
.c11:focus {
  background: rgba(0,114,165,1);
}

.c11:active {
  background: rgba(0,114,165,1);
}

.c11:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.c6 {
  padding: 0;
  position: relative;
}

.c7 {
  border: 0;
  border-bottom: 1px solid rgba(227,232,236,1);
  font: 400 13px/ 1 'Lato';
  outline: none;
  padding: 8px 8px 16px 24px;
  width: 100%;
}

.c7::selection {
  background: rgba(0,129,187,1);
  color: white;
}

.c7::-webkit-input-placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7::-moz-placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7:-ms-input-placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7::placeholder {
  color: rgba(11,26,52,0.5);
  font: 400 13px/ 1 'Lato';
}

.c7:focus {
  color: rgba(0,129,187,1);
  font-weight: 400;
}

.c8 {
  bottom: 15px;
  color: rgba(60,79,114,0.6);
  position: absolute;
  z-index: 1;
}

.c0 {
  background: white;
  border-left: 1px solid rgba(216,216,216,0.8);
  border-right: 1px solid rgba(216,216,216,0.8);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 336px;
  overflow: hidden;
  position: relative;
}

.c5 {
  position: relative;
}

.c4 {
  background: transparent;
  border: 0 none;
  color: rgba(11,26,52,1);
  cursor: pointer;
  display: inline-block;
  font: 300 13px/1 'Lato';
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.3;
  outline: none;
  padding: 16px 0 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: opacity 0.195s cubic-bezier(0,0,0.2,1);
  transition: opacity 0.195s cubic-bezier(0,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: auto;
  z-index: 1;
}

.c4:hover {
  color: rgba(0,129,187,1);
  opacity: 1;
}

.c4:disabled {
  opacity: 0;
  pointer-events: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  opacity: 1;
  overflow: hidden;
  position: relative;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 0.195s cubic-bezier(0.4,0,1,1);
  -webkit-transition: transform 0.195s cubic-bezier(0.4,0,1,1);
  transition: transform 0.195s cubic-bezier(0.4,0,1,1);
  width: 672px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  position: relative;
  width: 336px;
}

.c3 {
  -webkit-flex: 0 0 72px;
  -ms-flex: 0 0 72px;
  flex: 0 0 72px;
  margin: 0 16px;
  min-height: 72px;
  padding: 24px 0 0;
  position: relative;
}

.c9 {
  background: white;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  margin: 0;
  overflow: auto;
  padding: 16px 0;
  position: relative;
  width: 100%;
}

.c10 {
  background: white;
  -webkit-flex: 0 0 72px;
  -ms-flex: 0 0 72px;
  flex: 0 0 72px;
  margin: auto 0 0;
  min-height: 72px;
  padding: 16px;
  width: 100%;
}

.c12 {
  -webkit-flex: 0 0 336px;
  -ms-flex: 0 0 336px;
  flex: 0 0 336px;
  overflow: hidden;
  max-width: 336px;
  position: relative;
  width: 336px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <header
        className="c3"
      >
        <button
          className="c4"
          onClick={undefined}
        >
          Clear all filters
        </button>
        <div
          className="c5"
        >
          <div
            className="c6"
          >
            <input
              className="c7"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
              value=""
            />
            <div
              className="c8"
            >
              <svg
                viewBox="0 0 17 17"
                width="15px"
              >
                <g
                  fill="none"
                  stroke="none"
                  strokeLinecap="round"
                  strokeWidth="1"
                >
                  <g
                    stroke="currentColor"
                    strokeWidth="2"
                    transform="translate(1.000000, 1.000000)"
                  >
                    <g>
                      <circle
                        cx="6.5"
                        cy="6.5"
                        r="6"
                      />
                      <path
                        d="M14.5,14.5 L10.742,10.742"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
      </header>
      <div
        className="c9"
      />
      <div
        className="c10"
      >
        <button
          className="c11"
          disabled={undefined}
          onClick={[Function]}
        >
          Done
        </button>
      </div>
    </div>
    <div
      className="c12"
    />
  </div>
</div>
`;
